src/main/java/budgetbuddy/storage/loans/JsonAdaptedLoan.java

```diff
@@ -66,91 +65,88 @@ public JsonAdaptedLoan(Loan source) {
      * @throws IllegalValueException If any data constraints were violated in the adapted loan.
      */
     public Loan toModelType() throws IllegalValueException {
-        checkName();
-        checkDirection();
-        checkAmount();
-        checkDate();
-        checkDescription();
-        checkStatus();
-
-        Person person = new Person(new Name(personName), new LoanList());
-        Loan loan = new Loan(person, Direction.valueOf(direction), new Amount(amount),
-                date, new Description(description), Status.valueOf(status));
-        person.addLoan(loan);
-        return loan;
+        Person person = new Person(checkName());
+        return new Loan(person, checkDirection(), checkAmount(),
+                checkDate(), checkDescription(), checkStatus());
```

Comment:
Should probably rename the `check*` methods now that they don't only check.

(TODO FIXME)

Reply:
Good point, I'll change it in a future PR.